src/row_codec.rs

```diff
@@ -13,25 +16,26 @@ bitflags! {
 const NO_EXPIRE_TS: i64 = i64::MIN;
 
 /// Encodes key and value using the binary codec for SlateDB row representation
-/// using the `v0` encoding scheme.
+/// using the `v1` encoding scheme.
 ///
-/// The `v0` codec for the key is (for non-tombstones):
+/// The `v1` codec for the key is (for non-tombstones):
 ///
 /// ```txt
-///  |--------------------------------------------------------------------------|
-///  |       u16      |      u16       |  var        | var  |    u32    |  var  |
-///  |----------------|----------------|-------------|------|-----------|-------|
-///  | key_prefix_len | key_suffix_len |  key_suffix | meta | value_len | value |
-///  |--------------------------------------------------------------------------|
+///  |-------------------------------------------------------------------------------------------------------------|
+///  |       u16      |      u16       |  var        | u32     | 1 byte    | u16       | var   |    u32    |  var  |
```

Comment:
Is it better to use `u64` instead of `u32` for seqnum?

Reply:
it seems that leveldb used a 7 bytes sequence number, and they claimed that 56 bits is enough 😲:

> This shouldn't happen in any foreseeable time: let's assume that we are writing at 1M per second. That means we will run out of sequence numbers after 2^56/1000000 seconds, which is over 2000 years.

https://github.com/google/leveldb/issues/792#issuecomment-606789474

Reply:
https://github.com/facebook/rocksdb/blob/24045549a61d755418d5e9e6b208b2bfd77093b8/db/dbformat.h#L128
RocksDB seems to use a similar approach, their seqnum uses 56 bits, and the rest is used for types. But I think u32 is not enough.🤔

Reply:
sorry my bad, i mis-calculated u32 as 8 bytes 🥲

thank you for reminding, i'll fix it

Reply:
fixed in 34a5c97